simtfl/network.py

```diff
@@ -30,16 +34,34 @@ def add_node(self, node):
         """
         Adds a node with the next available ident.
         """
+        ident = self.num_nodes()
         self.nodes.append(node)
+        node.initialize(ident, self.env, self)
 
     def start_node(self, ident):
         """
         (process) Start the node with the given ident.
         """
         node = self.node(ident)
-        print(f"T{self.env.now:5d}: starting  {node}")
+        print(f"T{self.env.now:5d}: starting  {ident:2d}: {node}")
         return node.run()
 
+    def start_processes(self):
+        """
+        Start a process for each node.
+        """
+        print()
+        for i in range(self.num_nodes()):
+            self.env.process(self.start_node(i))
```

Comment:
I understand this design, but I keep wondering why nodes don't hold both their identity and `env` so this could be written as:

```
for node in self.nodes:
    node.start()
```

Any thoughts about that trade-off?

Reply:
This would work now; it wouldn't have worked with the previous design. I'll change it.